src/json_schema_validation.rs

```diff
@@ -0,0 +1,204 @@
+use anyhow::{bail, Context, Result};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+
+// TODO: Consider using `Value` type from `serde_json`
```

Comment:
There's not an easy way to reuse that here, so I think it's fine to define this new enum